tests/MongoDB.Driver.Core.Tests/Core/Clusters/MultiServerClusterTests.cs

```diff
@@ -73,6 +100,18 @@ public void Constructor_should_throw_if_cluster_connection_mode_is_not_supported
             act.ShouldThrow<ArgumentException>();
         }
 
+        public void constructor_should_use_default_DnsMonitorFactory_when_dnsMonitorFactory_is_null()
+        {
+            var settings = new ClusterSettings();
+            var serverFactory = Mock.Of<IClusterableServerFactory>();
+            var eventSubscriber = Mock.Of<IEventSubscriber>();
+
+            var result = new MultiServerCluster(settings, serverFactory, eventSubscriber, dnsMonitorFactory: null);
+
+            var dnsMonitorFactory = result._dnsMonitorFactory().Should().BeOfType<DnsMonitorFactory>().Subject;
```

Comment:
Is this really asserting that it's the default, as the test name indicates?

Reply:
Yes to the best of our ability here.

We passed in null and we are asserting that the null was replaced with the default. We could have just checked that the value was not null, but it seems to be useful to also check that the type is actually that of the default implementation.